src/data/markdown/docs/30 xk6-browser/01 xk6-browser/01-browser.md

```diff
@@ -19,25 +23,108 @@ export default function () {
 
 
 ## browser.close()
 
+In case this browser is obtained using [browserType.launch([options])](03-browser-type.md#launch), closes the browser and all of its pages (if any were opened).
+
+In case this browser is connected to, clears all created contexts belonging to this browser and disconnects from the browser server.
+
+The [Browser](01-browser.md) object itself is considered to be disposed and cannot be used anymore.
+
 ## browser.contexts()
 
+Returns an array of all open browser contexts. In a newly created browser, this will return zero browser contexts.
+
+<!-- eslint-skip -->
+
+```javascript
+const browser = launcher.launch('chromium');
+console.log(browser.contexts().length); // prints `0`
+
+const context = browser.newContext();
+console.log(browser.contexts().length); // prints `1`
+```
+
 ## browser.isConnected()
 
+Indicates that the browser is connected.
+
 ## browser.newBrowserCDPSession()
 
+Returns the newly created browser session.
+
+> **Note**:
+> CDP Sessions are only supported on Chromium-based browsers.
+
 ## browser.newContext([options])
 
+### options: returns <[Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object)>
+
+You can customize the creation of a new browser context using the following options.
+
+<!-- vale off -->
+
+| Option            | Type                   | Description |
+| ----------------- | :--------------------- | :---------- |
+| acceptDownloads   | boolean                | TODO |
+| bypassCSP         | boolean                | TODO |
+| colorScheme       | string                 | Emulates 'prefers-colors-scheme' media feature. It can be one of `"light"`, `"dark"`, `"no-preference"`. |
+| deviceScaleFactor | number                 | TODO |
+| extraHTTPHeaders  | object<string, string> | TODO |
+| geolocation       | object                 | See [geolocation](#newcontext-geolocation). |
+| hasTouch          | boolean                | TODO |
+| httpCredentials   | object                 | Credentials for HTTP authentication. See: [httpCredentials](#newcontext-http-credentials). |
+| ignoreHTTPSErrors | boolean                | TODO |
+| isMobile          | boolean                | TODO |
+| javaScriptEnabled | boolean                | TODO |
+| locale            | string                 | TODO |
+| offline           | boolean                | TODO |
+| permissions       | Array<string>          | TODO |
+| reducedMotion     | string                 | TODO |
+| screen            | object                 | Emulates consistent window screen size available inside web page via window.screen. Is only used when the viewport is set. See: [screen](#newcontext-screen) |
+| timezoneID        | xx                     | TODO |
+| userAgent         | xx                     | TODO |
+| videosPath        | xx                     | TODO |
+| viewport          | xx                     | TODO |
+
+<!-- vale on -->
+
+### <a name="newcontext-geolocation">geolocation option</a>
```

Comment:
Ugh yeah, this will get messy. :disappointed:

Playwright's nested unordered list is more compact, but we need to have tables for everything... Nested tables would be more compact, but even messier, and we couldn't link to them.

So I guess this is fine :man_shrugging:, but I'd really like to see it rendered.

Reply:
Yeah, don't trust a standalone Markdown preview for this. The Gatsby flavor and our own changes is a separate beast entirely... :sweat: